src/libraries/System.Text.Json/gen/JsonSourceGenerator.cs

```diff
@@ -67,7 +67,7 @@ private sealed class SyntaxReceiver : ISyntaxReceiver
 
             public void OnVisitSyntaxNode(SyntaxNode syntaxNode)
             {
-                if (syntaxNode is ClassDeclarationSyntax cds)
+                if (syntaxNode is ClassDeclarationSyntax { AttributeLists.Count: > 0, BaseList.Types.Count: > 0 } cds)
```

Comment:
Can we further constrain it to `partial` classes or would that loose a valuable diagnostic in the case the user forgot to mark the class `partial`?  I guess we're already losing a diagnostic here when the user forgot to derive from `JsonSerializerContext`.  Do we care?

Reply:
Yeah, I considered that. It does seem odd that it only warns on missing `partial`. 

Aside: is there a reason as a user that I can't just put an attribute on my serialization class, and have it generate the whole thing for me. Seems weird I have to make an empty partial class.

Reply:
> Aside: is there a reason as a user that I can't just put an attribute on my serialization class, and have it generate the whole thing for me.

What name and namespace would we generate for the class that derives from JsonSerializerContext?

There is also the scenario of when you want to serialize a Type in an assembly that you don't own/develop.

Reply:
> What name and namespace would we generate for the class that derives from JsonSerializerContext? 

I mean, you could supply it in the attribute, or derive it. 

> There is also the scenario of when you want to serialize a Type in an assembly that you don't own/develop.

Assembly attribute? But I digress. I just found the experience a little odd.

Reply:
We welcome the feedback. If you have an idea how we can improve the experience, we'd like to hear it.

Reply:
We actually used to have assembly level attributes for interacting with the generator and then changed course to the partial class.  Some of that discussion is here: https://github.com/dotnet/runtime/issues/51945#issuecomment-841011091